src/components/parameters/ParameterInfo.svelte

```diff
@@ -0,0 +1,114 @@
+<svelte:options immutable={true} />
+
+<script lang="ts">
+  import InfoIcon from '@nasa-jpl/stellar/icons/info.svg?component';
+  import { createEventDispatcher } from 'svelte';
+  import type { FormParameter, ValueSource } from '../../types/parameter';
+  import ContextMenu from '../context-menu/ContextMenu.svelte';
+  import ValueSourceBadge from './ValueSourceBadge.svelte';
+
+  export let formParameter: FormParameter;
+
+  const dispatch = createEventDispatcher();
+
+  let contextMenu: ContextMenu;
+  let isIconHovered: boolean = false;
+  let isTooltipHovered: boolean = false;
+  let leaveTimeout: NodeJS.Timeout | null = null;
+  let source: ValueSource;
+  let unit: string | undefined = undefined;
+
+  $: if (formParameter) {
+    source = formParameter.valueSource;
+    unit = formParameter.schema?.metadata?.unit?.value;
+  }
+
+  function leaveCallback() {
+    if (leaveTimeout != null) {
+      clearTimeout(leaveTimeout);
+    }
+
+    leaveTimeout = setTimeout(() => {
+      if (isIconHovered || isTooltipHovered) {
+        leaveCallback();
+      } else {
+        contextMenu.hide();
+      }
+    }, 300);
+  }
+
+  function hoverCallback() {
+    if (leaveTimeout != null) {
+      clearTimeout(leaveTimeout);
+    }
+  }
+
+  function onIconOver(e: MouseEvent) {
+    isIconHovered = true;
+    if (!contextMenu.isShown()) {
+      const bounds = (e.target as HTMLElement).getBoundingClientRect();
+      contextMenu.showDirectly(bounds.x + bounds.width, bounds.y, bounds.x);
+    }
+    hoverCallback();
+  }
+
+  function onReset() {
+    dispatch('reset', formParameter);
+  }
+
+  function onTooltipOver() {
+    isTooltipHovered = true;
+    hoverCallback();
+  }
+
+  function onIconOut() {
+    isIconHovered = false;
+    leaveCallback();
+  }
+
+  function onTooltipOut() {
+    isTooltipHovered = false;
+    leaveCallback();
+  }
+</script>
+
+{#if unit || source !== 'none'}
+  <div class="parameter-info-container" role="contentinfo" on:mouseenter={onIconOver} on:mouseleave={onIconOut}>
+    <div><InfoIcon /></div>
+    <ContextMenu hideAfterClick={false} bind:this={contextMenu}>
```

Comment:
Sometimes seeing this interesting behavior where conte

Uploading Screen Recording 2023-11-02 at 8.25.10 AM.mov…

xt menu doesn't always clear on hover out

Reply:
Got it (I think). Is there any way to differentiate between this situation and normal usage where you are trying to go from icon -> tooltip?

Reply:
@AaronPlave I've lowered the timeout to hopefully reduce the chances of this happening. I also added a transition delay to the hover event on the parent component to sync up with the tooltip's mouse out timeout

Reply:
Fixes the issue though now I guess we might have to live with this sort of side effect?
https://github.com/NASA-AMMOS/aerie-ui/assets/4419607/d618b151-51ce-438b-a107-8e7516cd9a58